packages/video_player/video_player/lib/video_player.dart

```diff
@@ -399,6 +399,12 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
         case VideoEventType.bufferingEnd:
           value = value.copyWith(isBuffering: false);
           break;
+        case VideoEventType.playingUpdate:
+          // TODO(maRci002): make VideoPlayerValue immutable and provide equals / hash
```

Comment:
What will the code be like then?

Reply:
Currently I'm doing something like this to avoid unecessary `VideoPlayer` widget renders:

```dart
  if (event.isPlaying! != value.isPlaying) {
	value = value.copyWith(isPlaying: event.isPlaying);
  }
```

However since `VideoPlayerController extends ValueNotifier<VideoPlayerValue>` is a `ValueNotifier` then `VideoPlayerValue` could support equals / hash methods, I'm going to add it.

[ValueNotifier](https://api.flutter.dev/flutter/foundation/ValueNotifier-class.html):

> A [ChangeNotifier](https://api.flutter.dev/flutter/foundation/ChangeNotifier-class.html) that holds a single value.
> 
> When [value](https://api.flutter.dev/flutter/foundation/ValueNotifier/value.html) is replaced with something that is not equal to the old value as evaluated by the equality operator ==, this class notifies its listeners.

Reply:
I made `VideoPlayerValue`, `Caption` immutable